src/org/omegat/gui/theme/DefaultFlatTheme.java

```diff
@@ -44,113 +43,71 @@
-     * well known look and feel implementations.  Presently
-     * the well known names are "Motif", "Windows", "Mac", "Metal".  Note
-     * that a LookAndFeel derived from a well known superclass
-     * that doesn't make any fundamental changes to the look or feel
-     * shouldn't override this method.
-     *
-     * @return identifier for the look and feel
-     */
     @Override
     public String getID() {
-        return "Default";
+        return "OmegaT"; // NOI18N
     }
 
-    /**
-     * Return a one line description of this look and feel implementation,
-     * e.g. "The CDE/Motif Look and Feel".   This string is intended for
-     * the user, e.g. in the title of a window or in a ToolTip message.
-     *
-     * @return short description for the look and feel
-     */
     @Override
     public String getDescription() {
-        return DESCRIPTION;
+        return OStrings.getString("THEME_OMEGAT_DEFAULT_DESCRIPTION");
     }
 
-    /**
-     * If the underlying platform has a "native" look and feel, and
-     * this is an implementation of it, return {@code true}.  For
-     * example, when the underlying platform is Solaris running CDE
-     * a CDE/Motif look and feel implementation would return {@code
-     * true}.
-     *
-     * @return {@code true} if this look and feel represents the underlying
-     * platform look and feel
-     */
     @Override
     public boolean isNativeLookAndFeel() {
-        return false;
+        return systemLookAndFeel.isNativeLookAndFeel();
     }
 
-    /**
-     * Return {@code true} if the underlying platform supports and or permits
-     * this look and feel.  This method returns {@code false} if the look
-     * and feel depends on special resources or legal agreements that
-     * aren't defined for the current platform.
-     *
-     * @return {@code true} if this is a supported look and feel
-     * @see UIManager#setLookAndFeel
-     */
     @Override
     public boolean isSupportedLookAndFeel() {
-        return true;
+        return systemLookAndFeel.isSupportedLookAndFeel();
     }
 
+    @Override
     public UIDefaults getDefaults() {
-        UIDefaults defaults = systemLookAndFeel.getDefaults();
+        // Use system LAF ID because we use the ID to see if it's e.g. Windows
+        return setDefaults(super.getDefaults(), systemLookAndFeel.getID());
+    }
+
+    /**
+     * Apply theme default values to the supplied {@link UIDefaults} object.
+     * Suitable for third-party themes that want to modify the default theme.
+     *
+     * @param defaults the {@link UIDefaults} object such as from
+     *                 {@link LookAndFeel#getDefaults()}
+     * @param lafId    the ID of the LAF that will make use of the defaults (from
+     *                 {@link LookAndFeel#getID()})
+     * @return the modified {@link UIDefaults} object
+     */
+    public static UIDefaults setDefaults(UIDefaults defaults, String lafId) {
```

Comment:
I'd like to try using `setDefaults(...)` from sample 3rd-party plugin, and give a feedback here.

Reply:
There are 3 feedbacks from 3rd-party plugin implementation,

1. UIDesignManager#initialize should call `loadColors()` after `setTheme` because it fails to detect dark custom theme, then it load light color sheme even when dark theme applied.
2. Second argument `lafId` of `setDefaults` should be changed. It is used to detect OS(Windows) but  custom LaF name does not work for it.
3. 3rd-party LaF also need to use `UIManager.put` for VLDocking related customize. 
`        UIManager.put("DockViewTitleBar.border", new MatteBorder(1, 1, 1, 1, borderColor));`


![image](https://user-images.githubusercontent.com/123720/121792176-937cf780-cc2c-11eb-8db0-5e8035ae27a2.png)

Reply:
There is future improvement space in `loadColors`. Current implementation only have 2 fixed defaults, light and dark.
Because it depends on theme color scheme,  we should allow theme provider to customize colors.

When we or plugin developer provide a high-contrast theme for weak-eyed users, it is going to be a big problem.

Reply:
When `setDefaults` param `lafId` intend to accept `systemLookAndFeel`  in any case, eg. even in 3rd-paty plugins, it is not necessary to get it as argument.

Reply:
> There are 3 feedbacks from 3rd-party plugin implementation,

Thanks for trying it out.

>     1. UIDesignManager#initialize should call `loadColors()` after `setTheme` because it fails to detect dark custom theme, then it load light color sheme even when dark theme applied.

Thanks for making this change.

>     2. Second argument `lafId` of `setDefaults` should be changed. It is used to detect OS(Windows) but  custom LaF name does not work for it.

I'm not sure what we should do with this. It's not detecting Windows the OS; it's detecting a specific Windows LAF. For other LAFs I think it's OK to pass the LAF ID and it simply won't recognize it.

>     3. 3rd-party LaF also need to use `UIManager.put` for VLDocking related customize.
>        `        UIManager.put("DockViewTitleBar.border", new MatteBorder(1, 1, 1, 1, borderColor));`

Yes. Until we can improve VLDocking to make it possible for it to install its defaults in the LAF defaults, we will need to document that third-party LAFs must use this workaround.

Reply:
> I'm not sure what we should do with this. It's not detecting Windows the OS; it's detecting a specific Windows LAF. For other LAFs I think it's OK to pass the LAF ID and it simply won't recognize it.

It is now very clear. I've satisfied the PR 100%.